[docs, API] control api + swagger ui for http api (#3158)
GoogleContainerTools/skaffold: docs/content/en/docs/design/api.md

To retrieve information about the Skaffold pipeline, the Skaffold API provides two main functionalities:
  

>   * A [streaming event log]({{< relref "#events-api">}}) created from the different phases in a pipeline run and

Reviewer: ```suggestion
  * A [streaming event log]({{< relref "#events-api">}}) created from the different phases in a pipeline run, and
```

  
  * A snapshot of the [overall state]({{< relref "#state-api" >}}) of the pipeline at any given time during the run.

> To control the individual phases of the Skaffold, the Skaffold API provides [fine grained control over]({{< relref "#controlling-build-sync-deploy" >}})

Reviewer: ```suggestion
To control the individual phases of the Skaffold, the Skaffold API provides [fine grained control]({{< relref "#controlling-build-sync-deploy" >}}) over
```

Author: I don't get this change

Reviewer 2: this is very small nit. i am going to approve and merge

The server is hosted locally on the same host where the skaffold process is running, and will serve by default on ports 50051 and 50052.
These ports can be configured through the `--rpc-port` and `--rpc-http-port` flags.

> We generate the server's [gRPC service definitions and message protos]({{< relref "/docs/references/api/grpc" >}}) as well as the [Swagger based HTTP API Spec]({{< relref "/docs/references/api/swagger" >}}).

Reviewer: ```suggestion
For reference, we generate the server's [gRPC service definitions and message protos]({{< relref "/docs/references/api/grpc" >}}) as well as the [Swagger based HTTP API Spec]({{< relref "/docs/references/api/swagger" >}}).
```

```
{{% /tab %}}
{{% tab "gRPC API" %}}

> To get events from the `gRPC` server, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}})

Reviewer: ```suggestion
To get events from the API using `gRPC`, first create a [`gRPC` client]({{< relref "#creating-a-grpc-client" >}}):
```

```golang
func main() {
  ctx, ctxCancel := context.WithCancel(context.Background())
  defer ctxCancel()
  // `client` is the gRPC client with connection to localhost:50051.
  // See code above to create it

>   logStream, err := client.EventLog(ctx, &empty.Empty{})

Reviewer: didn't we want to keep this as `client.Events`?

Author: yes we should, good catch

Author: ```suggestion
  logStream, err := client.Events(ctx, &empty.Empty{})
```

{{% /tab %}}
{{% /tabs %}}

> Each [Entry log]({{<relref "/docs/references/api/grpc#proto.LogEntry" >}}) contains an [Event]({{< relref "/docs/references/api/grpc#proto.Event" >}}) in the `LogEntry.Event` field and

Reviewer: ```suggestion
Each [Entry]({{<relref "/docs/references/api/grpc#proto.LogEntry" >}}) in the log contains an [Event]({{< relref "/docs/references/api/grpc#proto.Event" >}}) in the `LogEntry.Event` field and
```

```
{{% /tab %}}
{{% tab "gRPC API" %}}

> To get events over `gRPC` server, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}})

Reviewer: ```suggestion
To retrieve the state from the server using `gRPC`, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}}) as before:
```
